modulus/utils/capture.py

```diff
@@ -229,31 +230,74 @@ def _amp_forward(self, *args, **kwargs) -> Any:
             self.scaler.scale(output).backward()
         return output
 
-    @classmethod
-    def _register_scaler(
-        cls, scaler: torch.cuda.amp.GradScaler, logger: Logger
-    ) -> None:
-        """Class method for saving/loading the grad scaler state dictionary singleton
-
-        Parameters
-        ----------
-        scaler : torch.cuda.amp.GradScaler
-            AMP grad scaler
-        logger : Logger
-            Python console logger
-        """
-        if cls.scaler_dict:
+    def _init_amp_scaler(
+        self, scaler_enabled: bool, logger: Logger
+    ) -> torch.cuda.amp.GradScaler:
+        # Create gradient scaler
+        scaler = torch.cuda.amp.GradScaler(enabled=scaler_enabled)
+        # Store scaler in class variable
+        scaler_id = f"scaler_{len(_StaticCapture._amp_scalers.keys())}"
```

Comment:
Is this robust to permutations of the order in which models are captured in the training script? Wondering if it might be good to allow a user to pass in an optional label for the static capture

Reply:
Yeah right now its not... I had actually put an error because its order dependent (ignore the spelling error)

https://github.com/NickGeneva/modulus/blob/fix/static_capture/modulus/utils/capture.py#L254C68-L254C76

I suppose I figured that people arent typically moving stuff around during training and if you are odds are you'll want to implement all this yourself for direct control. Do you have a specific example in mind where you would have this occur?

Reply:
The only example where this might be an issue is if someone trains a model, then makes some changes to the training script but still wants to restart from the same checkpoint that they happen to know should work.

If we just add a `label : Optional[str] = None` kwarg to the constructor though, a user can set that and if they don't, then they're relying on this implicit naming convention?

Reply:
Added this with a test that checks the order in-variance does indeed work (test took longer than the feature).